### PR TITLE
Side menu

### PR DESCRIPTION
**Pull request by BrettBukowski**
- Adds a basic toolbar and places the hamburger + pane controls in it
- Adds side menu component that displays available chapters
- Adds chapter reducer + actions
- Adds menu state reducer + actions

*****

### CODE REVIEW

File: app/components/SideMenu.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	render() {
+    return (
+      <Animated.View scrollsToTop={false} style={[
+        styles.container, {
+          width: this.state.menuWidth,
+          left: this.state.menuLeft,
+        }
+      ]} {...this._panResponder.panHandlers}>
+        <View style={styles.menu}>
+          {this.props.chapters.map((chapter, i) => {
```
Comment (krapspark) [inline app/components/SideMenu.js]: I think this menu item component can be factored out into its own component if it ever gets bigger than what it is right now.
Fine for now.

File: app/components/Story.js [hunk 2]
```diff
@@ -2,34 +2,58 @@ import React, { Component } from 'react';
...
-    };
+    this.state = this._initializeDataSource();
 
     this.onClickBack = this.onClickBack.bind(this);
     this.onClickToggle = this.onClickToggle.bind(this);
     this._renderPage = this._renderPage.bind(this);
   }
 
+  _initializeDataSource(chapterId) {
+    let dataSource = new ViewPager.DataSource({ pageHasChanged: (r1, r2) => r1.text !== r2.text });
```
Comment (krapspark) [inline app/components/Story.js]: this can be const

File: app/components/Story.js [hunk 3]
```diff
@@ -2,34 +2,58 @@ import React, { Component } from 'react';
...
+
+    return {
+      data: dataSource.cloneWithPages(this._getChapterData(chapterId)),
+      isSplit: this.props.viewMode === 'pair',
```
Comment (krapspark) [inline app/components/Story.js]: this part isn't  part of initializing data source.
splitting out the data source is a good choice but i don't think all of state initialization belongs in it.

File: app/components/Story.js [hunk 4]
```diff
@@ -2,34 +2,58 @@ import React, { Component } from 'react';
...
-      isSplit: this.props.viewMode === 'pair'
-    };
+    this.state = this._initializeDataSource();
 
     this.onClickBack = this.onClickBack.bind(this);
     this.onClickToggle = this.onClickToggle.bind(this);
     this._renderPage = this._renderPage.bind(this);
   }
 
+  _initializeDataSource(chapterId) {
```
Comment (krapspark) [inline app/components/Story.js]: what's the heuristic for the function naming? they're all component functions so i'm not sure what the '_' means here

Comment (BrettBukowski) [inline app/components/Story.js]: Ah, that's a convention I've used writing react code elsewhere.. Basically the underscore denotes that it's a "custom" component method, making it easy to tell at a glance custom functionality from the react lifecycle methods in a component. I can remove the underscores.

File: app/components/SideMenu.js (hunk 1 above)
Comment (BrettBukowski) [inline app/components/SideMenu.js]: I went ahead and separated it out :)

File: app/components/Story.js (hunk 4 above)
Comment (BrettBukowski) [inline app/components/Story.js]: removed

**Comment by BrettBukowski** [comment]
alright, merging :)

**Pull request merged**